src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -1303,6 +1307,17 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 
 	public toggleChatInputOverlay(editing: boolean): void {
 		this.chatInputOverlay.classList.toggle('disabled', editing);
+		if (editing) {
+			this._overlayClickListener?.dispose();
+			this._overlayClickListener = this._register(dom.addStandardDisposableListener(this.chatInputOverlay, dom.EventType.CLICK, e => {
```

Comment:
You're registering it to the class and tracking the listener. This will have the class hold on to it after the listener is disposed.